src/RAMI.cpp

```diff
@@ -516,11 +518,12 @@ void RAMI::evalOp(const RamOperation& op, const InterpreterContext& args) {
             for (auto ip = range.first; ip != range.second; ++ip) {
                 const RamDomain* data = *(ip);
                 ctxt[scan.getTupleId()] = data;
-                visitSearch(scan);
+                if (!visitSearch(scan)) break;
```

Comment:
This is one of the safest one line if statements to have, but even so, for consistency wrap it with braces.